tests/Foundation/Bootstrap/RegisterFacadesTest.php

```diff
@@ -0,0 +1,150 @@
+use Illuminate\Foundation\AliasLoader;
+use Illuminate\Foundation\Application;
+use Illuminate\Foundation\Bootstrap\RegisterFacades;
+use Illuminate\Foundation\PackageManifest;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades;
+use Illuminate\Support\Js;
+use Illuminate\Support\Str;
+use Mockery as m;
+use PHPUnit\Framework\TestCase;
+use ReflectionClass;
+
+class RegisterFacadesTest extends TestCase
+{
+    protected function setUp(): void
+    {
+        $this->app = m::mock(Application::class);
+
+        $this->config = new Config();
+
+        $this->app->shouldReceive('make')
+            ->with('config')
+            ->once()
+            ->andReturn($this->config);
+
+        $this->packageManifest = m::mock(PackageManifest::class);
+
+        $this->app->shouldReceive('make')
+            ->with(PackageManifest::class)
+            ->once()
+            ->andReturn($this->packageManifest);
+    }
+
+    protected function tearDown(): void
+    {
+        m::close();
+
+        tap(AliasLoader::getInstance(), function ($loader) {
+            $loader->setAliases([]);
+            $loader->setRegistered(false);
+        });
+    }
+
+    public function testDefaultAliasesAreAdded()
+    {
+        $this->packageManifest->shouldReceive('aliases')
+            ->once()
+            ->andReturn([]);
+
+        (new RegisterFacades())->bootstrap($this->app);
+
+        foreach ($this->getDefaultAliases() as $alias => $abstract) {
+            $this->assertTrue(class_exists($alias));
+
+            $reflection = new ReflectionClass($alias);
+
+            $this->assertSame($abstract, $reflection->getName());
+        }
+
+        $this->assertCount(38, AliasLoader::getInstance()->getAliases());
+    }
+
+    public function testCustomAliasesCanBeAdded()
+    {
+        $this->config->set('app.aliases', [
+            'MyApp' => Facades\App::class,
+        ]);
+
+        $this->packageManifest->shouldReceive('aliases')
+            ->once()
+            ->andReturn([]);
+
+        (new RegisterFacades())->bootstrap($this->app);
+
+        $this->assertTrue(class_exists('MyApp'));
+        $reflection = new ReflectionClass('MyApp');
+
+        $this->assertSame(Facades\App::class, $reflection->getName());
+
+        $this->assertCount(39, AliasLoader::getInstance()->getAliases());
+    }
+
+    public function testPackageManifestAliasesCanBeAdded()
+    {
+        $this->packageManifest->shouldReceive('aliases')
+            ->once()
+            ->andReturn([
+                'MyPackageApp' => Facades\App::class,
+            ]);
+
+        (new RegisterFacades())->bootstrap($this->app);
+
+        $this->assertTrue(class_exists('MyPackageApp'));
+        $reflection = new ReflectionClass('MyPackageApp');
+
+        $this->assertSame(Facades\App::class, $reflection->getName());
+
+        $this->assertCount(39, AliasLoader::getInstance()->getAliases());
+    }
+
+    protected function getDefaultAliases()
```

Comment:
Maybe this method should be part of `class RegisterFacades` so the array below does not have to be duplicated in tests?